### PR TITLE
Issue #4801: fix some violations from eleventh part of idea violations

### PR DESCRIPTION
**Pull request by Nimfadora**
Issue #4801

violations included:
```
TestMethodWithoutAssertion
TransientFieldNotInitialized
SynchronizationOnStaticField
UnnecessaryDefault
```

### CODE REVIEW

File: src/test/java/com/puppycrawl/tools/checkstyle/grammars/AstRegressionTest.java [hunk 1]
```diff
@@ -218,7 +218,7 @@ private static void verifyAstRaw(String expectedTextPrintFileName, String actual
                 actualContents);
     }
 
-    private static final class AssertGeneratedJavaLexer extends GeneratedJavaLexer {
+    public static final class AssertGeneratedJavaLexer extends GeneratedJavaLexer {
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/grammars/AstRegressionTest.java]: Why was this changed?

Comment (Nimfadora) [inline src/test/java/com/puppycrawl/tools/checkstyle/grammars/AstRegressionTest.java]: it contains test methods which are used in this class, so I add them to idea TestMethodWithoutAssertion violations test method list

Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/grammars/AstRegressionTest.java]: And it fails if the class is left `private`?

Comment (Nimfadora) [inline src/test/java/com/puppycrawl/tools/checkstyle/grammars/AstRegressionTest.java]: fixed

**Comment by romani** [comment]
manually mereged
